app/Models/UserQuery.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/**
+ * Contains the description of a user query
+ * 
+ * It allows to extract the meaningful bits of the query to be manipulated in an
+ * easy way.
+ */
+class FreshRSS_UserQuery {
+
+	private $deprecated = false;
+	private $get;
+	private $get_name;
+	private $get_type;
+	private $name;
+	private $order;
+	private $search;
+	private $state;
+	private $url;
+	private $feed_dao;
+	private $category_dao;
+
+	/**
+	 * @param array $query
+	 * @param FreshRSS_Searchable $feed_dao
+	 * @param FreshRSS_Searchable $category_dao
+	 */
+	public function __construct($query, FreshRSS_Searchable $feed_dao = null, FreshRSS_Searchable $category_dao = null) {
+		$this->category_dao = $category_dao;
+		$this->feed_dao = $feed_dao;
+		if (isset($query['get'])) {
+			$this->parseGet($query['get']);
+		}
+		if (isset($query['name'])) {
+			$this->name = $query['name'];
+		}
+		if (isset($query['order'])) {
+			$this->order = $query['order'];
+		}
+		if (!isset($query['search'])) {
+			$query['search'] = '';
+		}
+		// linked to deeply with the search object, need to use dependency injection
+		$this->search = new FreshRSS_Search($query['search']);
+		if (isset($query['state'])) {
+			$this->state = $query['state'];
+		}
+		if (isset($query['url'])) {
+			$this->url = $query['url'];
+		}
+	}
+
+	/**
+	 * Convert the current object to an array.
+	 * 
+	 * @return array
+	 */
+	public function toArray() {
+		return array_filter(array(
+		    'get' => $this->get,
+		    'name' => $this->name,
+		    'order' => $this->order,
+		    'search' => $this->search->__toString(),
+		    'state' => $this->state,
+		    'url' => $this->url,
+		));
+	}
+
+	/**
+	 * Parse the get parameter in the query string to extract its name and
+	 * type
+	 * 
+	 * @param string $get
+	 */
+	private function parseGet($get) {
+		$this->get = $get;
+		if (preg_match('/(?P<type>[acfs])(_(?P<id>\d+))?/', $get, $matches)) {
+			switch ($matches['type']) {
+				case 'a':
+					$this->parseAll();
+					break;
+				case 'c':
+					$this->parseCategory($matches['id']);
+					break;
+				case 'f':
+					$this->parseFeed($matches['id']);
+					break;
+				case 's':
+					$this->parseFavorite();
+					break;
+			}
+		}
+	}
+
+	/**
+	 * Parse the query string when it is an "all" query
+	 */
+	private function parseAll() {
+		$this->get_name = 'all';
+		$this->get_type = 'all';
+	}
+
+	/**
+	 * Parse the query string when it is a "category" query
+	 * 
+	 * @param integer $id
+	 */
+	private function parseCategory($id) {
+		$category = $this->category_dao->searchById($id);
```

Comment:
You should test `category_dao` is not null since it is optional in the constructor (i.e. default value at null)